src/sql/sqlitetypes.h

```diff
@@ -453,11 +458,13 @@ class Table : public Object
 private:
     QStringList fieldList() const;
     bool hasAutoIncrement() const;
+    std::shared_ptr<PrimaryKeyConstraint> getPk() const;
 
 private:
     QString m_rowidColumn;
     ConstraintMap m_constraints;
     QString m_virtual;
+    mutable QString m_onConflictClause;
```

Comment:
Why have this variable? As far as I can see the only places where it is accessed are ```setOnConflictClause()``` and ```onConflictClause``` (besides the assignment and the comparison operators). And in there the values are just forwarded from and to the primary key object. At least I feel like readability and maintainability would profit from removing this variable and using a local variable instead in both functions.

Reply:
Hi, @MKleusberg! Thanks for review!
I added variable m_onConflictClause for correct comparison of two Tables. 
Current Table::operator= copies pointers on Constrains from rhs.m_constraints to this->m_constraints, so primary key from "this" and primary key from rhs point on the same object.
```suggestion
Table& Table::operator=(const Table& rhs)
{
    ...
    m_constraints = rhs.m_constraints;
    ...
}
```
Table::operator== executing comparison of the primary keys can't recognize that PK has been changed.

So if I wouldn't use the variable I had to provide deep copy for each constraint of a table.

I can remove variable m_onConflictClause and create copy constructor for class Constraint and classes inherited from Constraint.

Could you please confirm that deep coping of collection m_constraints in Table::operator= is ok?